xarray/core/groupby.py

```diff
@@ -380,6 +384,9 @@ def __init__(
         # cached attributes
         self._groups = None
 
+        example = obj.isel(**{group_dim: group_indices[0]})
+        self.dims = example.dims
```

Comment:
made `dims` public because I thought it was a useful attribute.

Reply:
It might be worth creating a `dims` property to avoid eagerly slicing into obj:

```python
@property
def dims(self):
    if self._dims is None:
        self._dims = self._obj.isel(**{self._group_dim: self._group_indices[0]}).dims
    return self._dims
```

Reply:
Done.